AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginIntegrationTests/TransformerV2/DataStoreSchemaDriftTests.swift

```diff
@@ -0,0 +1,130 @@
+                                                  responseType: request.responseType,
+                                                  decodePath: request.decodePath,
+                                                  options: request.options)
+        Amplify.API.mutate(request: requestWithEnumThree) { result in
+            switch result {
+            case .success(let response):
+                print("\(response)")
+                switch response {
+                case .success(let result):
+                    XCTFail("should have failed decoding \(result)")
+
+                case .failure(let error):
+                    switch error {
+                    case .transformationError(let graphQLResponse, _):
+                        XCTAssertTrue(graphQLResponse.contains("enumValue\":\"THREE"))
+                        saveSuccessWithTransformationError.fulfill()
+                    default:
+                        XCTFail("Unexpected GraphQL response error \(error)")
+                    }
+                }
+            case .failure(let error):
+                XCTFail("Couldn't save with enum case `three`: \(error)")
+            }
+        }
+
+        wait(for: [saveSuccessWithTransformationError], timeout: TestCommonConstants.networkTimeout)
+
+        let dataStoreStartSuccess = expectation(description: "DataStore start success")
+        Amplify.DataStore.start { result in
+            if case .failure(let error) = result {
+                XCTFail("\(error)")
+            }
+            dataStoreStartSuccess.fulfill()
+        }
+        wait(for: [dataStoreStartSuccess], timeout: TestCommonConstants.networkTimeout)
+
+        // Assert that the sync engine does not retry on schema drift scenario
+        guard let remoteSyncEngine = DataStoreInternal.getRemoteSyncEngine() else {
+            XCTFail("Couldn't get RemoteSyncEngine, this could mean the internals have changed.")
+            return
+        }
+        let syncEngineCleanedUp = expectation(description: "SyncEngine cleaned up")
+        let syncEngineFailed = expectation(description: "SyncEngine failed")
+        let syncEngineRestarting = expectation(description: "SyncEngine restarting (this should not happen)")
+        syncEngineRestarting.isInverted = true
+        remoteSyncEngine.publisher.sink { completion in
+            switch completion {
+            case .finished:
+                break
+            case .failure:
+                syncEngineFailed.fulfill()
+            }
+        } receiveValue: { event in
+            if case .cleanedUp = event {
+                syncEngineCleanedUp.fulfill()
+            }
+            if case .schedulingRestart = event {
+                syncEngineRestarting.fulfill()
+            }
+        }.store(in: &subscriptions)
+        wait(for: [syncEngineCleanedUp, syncEngineFailed], timeout: TestCommonConstants.networkTimeout)
+        wait(for: [syncEngineRestarting], timeout: 10)
```

Comment:
is there a reason for keeping the timeout here different than `TestCommonConstants.networkTimeout`

Reply:
i don't think so, just old code, updating to use test constant